src/plugins/intel_npu/src/backend/src/zero_init.cpp

```diff
@@ -20,73 +20,34 @@ namespace intel_npu {
-    uint32_t targetVersion = 0;
-    for (uint32_t i = 0; i < count; ++i) {
-        auto& property = extProps[i];
-
-        if (strncmp(property.name, ZE_GRAPH_EXT_NAME, strlen(ZE_GRAPH_EXT_NAME)) != 0) {
-            continue;
-        }
-
-        // If the driver version is latest, will just use its name.
-        if (property.version == ZE_GRAPH_EXT_VERSION_CURRENT) {
-            graphExtName = property.name;
-            targetVersion = property.version;
-            break;
-        }
-
-        // Use the latest version supported by the driver.
-        if (property.version > targetVersion) {
-            graphExtName = property.name;
-            targetVersion = property.version;
-        }
-    }
-
-    if (graphExtName == nullptr) {
-        OPENVINO_THROW("queryDriverExtensionVersion: Failed to find Graph extension in NPU Driver");
-    }
-
-    const uint16_t supportedDriverExtMajorVersion = 1;
-    const uint16_t driverExtMajorVersion = ZE_MAJOR_VERSION(targetVersion);
-    if (supportedDriverExtMajorVersion != driverExtMajorVersion) {
-        OPENVINO_THROW("Plugin supports only driver with extension major version ",
-                       supportedDriverExtMajorVersion,
-                       "; discovered driver extension has major version ",
-                       driverExtMajorVersion);
-    }
-
-    return std::make_tuple(targetVersion, graphExtName);
-}
-
-static uint32_t queryMutableCommandListVersion(std::vector<ze_driver_extension_properties_t>& extProps,
-                                               uint32_t count) {
-    // query the mutable command list properties
+static std::tuple<uint32_t, std::string> queryDriverExtVersion(const char* extName,
+                                                               uint32_t expCurrentVersion,
+                                                               std::vector<ze_driver_extension_properties_t>& extProps,
+                                                               uint32_t count) {
+    const char* functionExtName = nullptr;
     uint32_t targetVersion = 0;
 
     for (uint32_t i = 0; i < count; ++i) {
         auto& property = extProps[i];
 
-        if (strncmp(property.name, ZE_MUTABLE_COMMAND_LIST_EXP_NAME, strlen(ZE_MUTABLE_COMMAND_LIST_EXP_NAME)) != 0) {
+        if (strncmp(property.name, extName, strlen(extName)) != 0) {
             continue;
         }
 
-        if (property.version == ZE_MUTABLE_COMMAND_LIST_EXP_VERSION_CURRENT) {
+        if (property.version == expCurrentVersion) {
+            functionExtName = property.name;
             targetVersion = property.version;
             break;
         }
 
         // Use the latest version supported by the driver.
         if (property.version > targetVersion) {
+            functionExtName = property.name;
             targetVersion = property.version;
         }
     }
 
-    return targetVersion;
+    return std::make_tuple(targetVersion, functionExtName ? functionExtName : "");
```

Comment:
no, we need to keep backward comparibility

Reply:
also, we want to create the pointer with version 0 if needed for further checks.

Reply:
Once error is detected, [it is reported](https://github.com/openvinotoolkit/openvino/blob/8f39bd37f6e8c96c5db1c7effbc4519405949cd8/src/plugins/intel_npu/src/backend/src/zero_init.cpp#L50) 

The same way should still be applied, without fake results it is more reliable.
Should the error be not fatal for the execution, an exception can be caught

Reply:
it doesn't make sense to add an extra mechanism of throw and catch for this.

Reply:
yeah, this is for an extension. But this is not the only one, we are checking some others as well. In case of them is not an error not finding it! The feature is just not available there!

Reply:
Then returning an `optional` is correct way to go here.

Reply:
an optional may be an option, but even so, we are returning a value there for the version and empty for the name.

Reply:
Two defaults are returned from auxiliary function in case of unsuccessful search within the loop. Both defaults are defined inside the auxiliary function and then leak out to the client calls of that function.
Any change in these defaults affect all the clients in a way they can even not be aware of. Just one call actually checks the result.
That is where an optional is clear choice as soon as failure to find the value is not the error for all the clients.